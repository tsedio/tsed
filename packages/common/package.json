{
  "name": "@tsed/common",
  "version": "6.63.2",
  "description": "A TypeScript Framework on top of Express",
  "source": "./src/index.ts",
  "main": "./lib/index.js",
  "typings": "./lib/index.d.ts",
  "private": false,
  "keywords": [
    "Express",
    "Koa",
    "TypeScript",
    "typescript",
    "Decorator",
    "decorators",
    "decorator",
    "express",
    "koa",
    "Controller",
    "Inject",
    "ioc",
    "di",
    "mvc",
    "swagger",
    "swagger ui",
    "ES2015",
    "ES6",
    "server",
    "rest",
    "api",
    "validation"
  ],
  "author": {
    "name": "Romain Lenzotti"
  },
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tsedio/tsed/issues"
  },
  "homepage": "http://tsed.io/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tsedio//tsed.git"
  },
  "scripts": {
    "build": "tsc --build tsconfig.compile.json"
  },
  "dependencies": {
    "@tsed/core": "6.63.2",
    "@tsed/perf": "6.63.2",
    "@tsed/di": "6.63.2",
    "@tsed/exceptions": "6.63.2",
    "@tsed/json-mapper": "6.63.2",
    "@tsed/logger": "^5.16.0",
    "@tsed/schema": "6.63.2",
    "@types/cache-manager": "^3.4.0",
    "@types/json-schema": "7.0.7",
    "cache-manager": "^3.4.1",
    "consolidate": "^0.16.0",
    "ejs": "^3.1.5",
    "json-schema": "^0.2.3",
    "on-finished": "2.3.0",
    "tslib": "2.2.0",
    "uuid": "8.3.2"
  },
  "devDependencies": {
    "@tsed/engines": "^1.1.4",
    "@types/consolidate": "0.14.0",
    "rxjs": "^6.5.2"
  },
  "peerDependencies": {
    "@tsed/engines": "^1.1.4"
  }
}