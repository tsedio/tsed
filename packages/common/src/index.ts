export * from "@tsed/di";
export * from "@tsed/platform-exceptions";
export * from "@tsed/platform-response-filter";
export * from "@tsed/platform-cache";
export * from "./config";
export * from "./mvc";
export * from "./platform";
export * from "./platform-test";

export {$log, Logger} from "@tsed/logger";
