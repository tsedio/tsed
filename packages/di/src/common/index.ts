/**
 * @file Automatically generated by @tsed/barrels.
 */
export * from "./constants/constants.js";
export * from "./decorators/autoInjectable.js";
export * from "./decorators/configuration.js";
export * from "./decorators/constant.js";
export * from "./decorators/controller.js";
export * from "./decorators/inject.js";
export * from "./decorators/injectable.js";
export * from "./decorators/intercept.js";
export * from "./decorators/interceptor.js";
export * from "./decorators/lazyInject.js";
export * from "./decorators/module.js";
export * from "./decorators/opts.js";
export * from "./decorators/overrideProvider.js";
export * from "./decorators/scope.js";
export * from "./decorators/service.js";
export * from "./decorators/useOpts.js";
export * from "./decorators/value.js";
export * from "./domain/Container.js";
export * from "./domain/ControllerProvider.js";
export * from "./domain/InjectablePropertyType.js";
export * from "./domain/LocalsContainer.js";
export * from "./domain/Provider.js";
export * from "./domain/ProviderScope.js";
export * from "./domain/ProviderType.js";
export * from "./errors/InjectionError.js";
export * from "./errors/InvalidPropertyTokenError.js";
export * from "./errors/UndefinedTokenError.js";
export * from "./interfaces/DIConfigurationOptions.js";
export * from "./interfaces/DILogger.js";
export * from "./interfaces/DILoggerOptions.js";
export * from "./interfaces/DIResolver.js";
export * from "./interfaces/ImportTokenProviderOpts.js";
export * from "./interfaces/InjectableProperties.js";
export * from "./interfaces/InterceptorContext.js";
export * from "./interfaces/InterceptorMethods.js";
export * from "./interfaces/InvokeOptions.js";
export * from "./interfaces/OnDestroy.js";
export * from "./interfaces/OnInit.js";
export * from "./interfaces/ProviderOpts.js";
export * from "./interfaces/RegistrySettings.js";
export * from "./interfaces/ResolvedInvokeOptions.js";
export * from "./interfaces/TokenProvider.js";
export * from "./interfaces/TokenRoute.js";
export * from "./registries/GlobalProviders.js";
export * from "./registries/ProviderRegistry.js";
export * from "./services/DIConfiguration.js";
export * from "./services/DILogger.js";
export * from "./services/InjectorService.js";
export * from "./utils/colors.js";
export * from "./utils/createContainer.js";
export * from "./utils/getConfiguration.js";
export * from "./utils/getConstructorDependencies.js";
export * from "./utils/resolveControllers.js";
