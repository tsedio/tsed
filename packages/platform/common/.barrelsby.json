{
  "directory": ["./src"],
  "exclude": ["**/__mock__", "**/__mocks__", "**/*.spec.ts", "**/getConfiguration.ts"],
  "delete": true
}
