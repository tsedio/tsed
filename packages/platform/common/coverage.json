{
  "statements": 92.25,
  "branches": 77.65,
  "functions": 88.81,
  "lines": 92.24
}
