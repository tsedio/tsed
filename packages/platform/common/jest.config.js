// For a detailed explanation regarding each configuration property, visit:
// https://jestjs.io/docs/en/configuration.html

module.exports = {
  ...require("@tsed/jest-config"),
  roots: ["<rootDir>/src", "<rootDir>/test"],
  moduleNameMapper: {
    "^@tsed/common$": "<rootDir>/src/index.ts"
  },
  coverageThreshold: {
    global: {
      statements: 98.05,
      branches: 90.32,
      functions: 97.08,
      lines: 98.05
    }
  }
};
