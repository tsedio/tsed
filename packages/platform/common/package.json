{
  "name": "@tsed/common",
  "version": "6.102.4",
  "description": "A TypeScript Framework on top of Express",
  "private": false,
  "source": "./src/index.ts",
  "main": "./lib/cjs/index.js",
  "typings": "./lib/types/index.d.ts",
  "exports": {
    "types": "./lib/types/index.d.ts",
    "import": "./lib/esm/index.js",
    "require": "./lib/cjs/index.js",
    "default": "./lib/esm/index.js"
  },
  "scripts": {
    "build": "yarn run build:esm && yarn run build:cjs",
    "build:cjs": "tsc --build tsconfig.compile.json",
    "build:esm": "tsc --build tsconfig.compile.esm.json"
  },
  "keywords": [
    "Express",
    "Koa",
    "TypeScript",
    "typescript",
    "Decorator",
    "decorators",
    "decorator",
    "express",
    "koa",
    "Controller",
    "Inject",
    "ioc",
    "di",
    "mvc",
    "swagger",
    "swagger ui",
    "ES2015",
    "ES6",
    "server",
    "rest",
    "api",
    "validation"
  ],
  "author": {
    "name": "Romain Lenzotti"
  },
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tsedio/tsed/issues"
  },
  "homepage": "http://tsed.io/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tsedio//tsed.git"
  },
  "funding": [
    {
      "type": "github",
      "url": "https://github.com/sponsors/Romakita"
    },
    {
      "type": "opencollective",
      "url": "https://opencollective.com/tsed"
    }
  ],
  "dependencies": {
    "@tsed/components-scan": "6.102.4",
    "@tsed/core": "6.102.4",
    "@tsed/di": "6.102.4",
    "@tsed/exceptions": "6.102.4",
    "@tsed/json-mapper": "6.102.4",
    "@tsed/logger": ">=6.0.2",
    "@tsed/logger-file": "^6.0.0",
    "@tsed/perf": "6.102.4",
    "@tsed/platform-cache": "6.102.4",
    "@tsed/platform-exceptions": "6.102.4",
    "@tsed/platform-log-middleware": "6.102.4",
    "@tsed/platform-middlewares": "6.102.4",
    "@tsed/platform-params": "6.102.4",
    "@tsed/platform-response-filter": "6.102.4",
    "@tsed/platform-views": "6.102.4",
    "@tsed/schema": "6.102.4",
    "@types/json-schema": "7.0.7",
    "@types/on-finished": "2.3.1",
    "on-finished": "2.3.0",
    "tslib": "2.2.0",
    "uuid": "8.3.2"
  },
  "devDependencies": {},
  "peerDependencies": {}
}
