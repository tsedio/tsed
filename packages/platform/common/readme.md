<p style="text-align: center" align="center">
 <a href="https://tsed.io" target="_blank"><img src="https://tsed.io/tsed-og.png" width="200" alt="Ts.ED logo"/></a>
</p>

<div align="center">
   <h1>@tsed/common</h1>

[![Build & Release](https://github.com/tsedio/tsed/workflows/Build%20&%20Release/badge.svg)](https://github.com/tsedio/tsed/actions?query=workflow%3A%22Build+%26+Release%22)
[![PR Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/tsedio/tsed/blob/master/CONTRIBUTING.md)
[![Coverage Status](https://coveralls.io/repos/github/tsedio/tsed/badge.svg?branch=production)](https://coveralls.io/github/tsedio/tsed?branch=production)
[![npm version](https://badge.fury.io/js/%40tsed%2Fcommon.svg)](https://badge.fury.io/js/%40tsed%2Fcommon)
[![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
[![code style: prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=flat-square)](https://github.com/prettier/prettier)
[![backers](https://opencollective.com/tsed/tiers/badge.svg)](https://opencollective.com/tsed)

</div>

<div align="center">
  <a href="https://tsed.io/">Website</a>
  <span>&nbsp;&nbsp;‚Ä¢&nbsp;&nbsp;</span>
  <a href="https://tsed.io/getting-started/">Getting started</a>
  <span>&nbsp;&nbsp;‚Ä¢&nbsp;&nbsp;</span>
  <a href="https://api.tsed.io/rest/slack/tsedio/tsed">Slack</a>
  <span>&nbsp;&nbsp;‚Ä¢&nbsp;&nbsp;</span>
  <a href="https://twitter.com/TsED_io">Twitter</a>
</div>

<hr />

> A Node.js and TypeScript Framework on top of Express. It provides a lot of decorators and guidelines to write your code.

## What it is

Ts.ED is a framework on top of Express that helps you to write your application in TypeScript (or in ES6). It provides a lot of decorators
to make your code more readable and less error-prone.

## Features

- Use our CLI to create a new project https://cli.tsed.io
- Support TypeORM, Mongoose, GraphQL, Socket.io, Swagger-ui, Passport.js, etc...
- Define class as Controller,
- Define class as Service (IoC),
- Define class as Middleware and MiddlewareError,
- Define class as Converter (POJ to Model and Model to POJ),
- Define root path for an entire controller and versioning your Rest API,
- Define as sub-route path for a method,
- Define routes on GET, POST, PUT, DELETE and HEAD verbs,
- Define middlewares on routes,
- Define required parameters,
- Inject data from query string, path parameters, entire body, cookies, session or header,
- Inject Request, Response, Next object from Express request,
- Template (View),
- Testing.

## Documentation

Documentation is available on [https://tsed.io](https://tsed.io)

## Getting started

See our [getting started here](https://tsed.io/getting-started/) to create new Ts.ED project or use
our [CLI](https://cli.tsed.io)

## Examples

Examples are available on [https://tsed.io/tutorials/](https://tsed.io/tutorials/)

## Overview

### Server example

Here an example to create a Server with Ts.ED:

```typescript
import {Configuration, Inject} from "@tsed/di";
import {PlatformApplication} from "@tsed/common";
import "@tsed/platform-express";
import Path from "path";
import cookieParser from "cookie-parser";
import compress from "compression";
import methodOverride from "method-override";

@Configuration({
  port: 3000,
  middlewares: [cookieParser(), compress({}), methodOverride()]
})
export class Server {}
```

To run your server, you have to use Platform API to bootstrap your application with the expected
platform like Express.

```typescript
import {$log} from "@tsed/common";
import {PlatformExpress} from "@tsed/platform-express";
import {Server} from "./Server";

async function bootstrap() {
  try {
    $log.debug("Start server...");
    const platform = await PlatformExpress.bootstrap(Server);

    await platform.listen();
    $log.debug("Server initialized");
  } catch (er) {
    $log.error(er);
  }
}

bootstrap();
```

To customize the server settings see [Configure server with decorator](https://tsed.io/docs/configuration.md)

#### Controller example

This is a simple controller to expose user resource. It use decorators to build the endpoints:

```typescript
import {Inject} from "@tsed/di";
import {Summary} from "@tsed/swagger";
import {
  Returns,
  ReturnsArray,
  Controller,
  Get,
  QueryParams,
  PathParams,
  Delete,
  Post,
  Required,
  BodyParams,
  Status,
  Put
} from "@tsed/common";
import {BadRequest} from "@tsed/exceptions";
import {UsersService} from "../services/UsersService";
import {User} from "../models/User";

@Controller("/users")
export class UsersCtrl {
  @Inject()
  protected usersService: UsersService;

  @Get("/:id")
  @Summary("Get a user from his Id")
  @Returns(User)
  async getUser(@PathParams("id") id: string): Promise<User> {
    return this.usersService.findById(id);
  }

  @Post("/")
  @Status(201)
  @Summary("Create a new user")
  @Returns(User)
  async postUser(@Required() @BodyParams() user: User): Promise<User> {
    return this.usersService.save(user);
  }

  @Put("/:id")
  @Status(201)
  @Summary("Update the given user")
  @Returns(User)
  async putUser(@PathParams("id") id: string, @Required() @BodyParams() user: User): Promise<User> {
    if (user.id !== id) {
      throw new BadRequest("ID mismatch with the given payload");
    }

    return this.usersService.save(user);
  }

  @Delete("/:id")
  @Summary("Remove a user")
  @Status(204)
  async deleteUser(@PathParams("id") @Required() id: string): Promise<User> {
    await this.usersService.delete(user);
  }

  @Get("/")
  @Summary("Get all users")
  @ReturnsArray(User)
  async findUser(@QueryParams("name") name: string) {
    return this.usersService.find({name});
  }
}
```

## Contributors

Please read [contributing guidelines here](./CONTRIBUTING.md).

<a href="https://github.com/tsedio/tsed/graphs/contributors"><img src="https://opencollective.com/tsed/contributors.svg?width=890" /></a>

## Backers

Thank you to all our backers! üôè [[Become a backer](https://opencollective.com/tsed#backer)]

<a href="https://opencollective.com/tsed#backers" target="_blank"><img src="https://opencollective.com/tsed/tiers/backer.svg?width=890"></a>

## Sponsors

Support this project by becoming a sponsor. Your logo will show up here with a link to your website. [[Become a sponsor](https://opencollective.com/tsed#sponsor)]

## License

The MIT License (MIT)

Copyright (c) 2016 - 2020 Romain Lenzotti

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

[travis]: https://travis-ci.org/
