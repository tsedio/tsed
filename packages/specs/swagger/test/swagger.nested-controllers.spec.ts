import {Controller, Get, PlatformTest, Post} from "@tsed/common";
import {PlatformExpress} from "@tsed/platform-express";
import SuperTest from "supertest";
import {Server} from "./app/Server";

@Controller("/")
export class FlaggedCommentController {
  @Get("/flag")
  async list() {}

  @Post("/:commentID/flag")
  async create() {}
}

@Controller({
  path: "/comments",
  children: [FlaggedCommentController]
})
export class CommentController {
  @Get("/")
  get() {}
}

@Controller({
  path: "/domain/:contextID",
  children: [CommentController]
})
export class DomainController {
  @Get("/")
  get() {}
}

describe("Swagger - nested controllers", () => {
  describe("OpenSpec3", () => {
    let request: SuperTest.SuperTest<SuperTest.Test>;
    beforeEach(
      PlatformTest.bootstrap(Server, {
        platform: PlatformExpress,
        mount: {
          "/rest": [DomainController, CommentController, FlaggedCommentController]
        }
      })
    );
    beforeEach(() => {
      request = SuperTest(PlatformTest.callback());
    });
    afterEach(PlatformTest.reset);

    it("should swagger spec", async () => {
      const response = await request.get("/v3/doc/swagger.json").expect(200);

      expect(response.body).toMatchSnapshot();
    });
  });
});
