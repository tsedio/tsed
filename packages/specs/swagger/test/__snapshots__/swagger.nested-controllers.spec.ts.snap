// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swagger - nested controllers OpenSpec3 should swagger spec 1`] = `
Object {
  "info": Object {
    "title": "Api documentation",
    "version": "1.0.0",
  },
  "openapi": "3.0.1",
  "paths": Object {
    "/rest/domain/{contextID}": Object {
      "get": Object {
        "operationId": "domainControllerGet",
        "parameters": Array [
          Object {
            "in": "path",
            "name": "contextID",
            "required": true,
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "DomainController",
        ],
      },
    },
    "/rest/domain/{contextID}/comments": Object {
      "get": Object {
        "operationId": "commentControllerGet",
        "parameters": Array [
          Object {
            "in": "path",
            "name": "contextID",
            "required": true,
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "CommentController",
        ],
      },
    },
    "/rest/domain/{contextID}/comments/flag": Object {
      "get": Object {
        "operationId": "flaggedCommentControllerList",
        "parameters": Array [
          Object {
            "in": "path",
            "name": "contextID",
            "required": true,
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "FlaggedCommentController",
        ],
      },
    },
    "/rest/domain/{contextID}/comments/{commentID}/flag": Object {
      "post": Object {
        "operationId": "flaggedCommentControllerCreate",
        "parameters": Array [
          Object {
            "in": "path",
            "name": "contextID",
            "required": true,
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "in": "path",
            "name": "commentID",
            "required": true,
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "FlaggedCommentController",
        ],
      },
    },
  },
  "tags": Array [
    Object {
      "name": "DomainController",
    },
    Object {
      "name": "CommentController",
    },
    Object {
      "name": "FlaggedCommentController",
    },
  ],
}
`;
